mlflow/tracing/types/wrapper.py

```diff
@@ -0,0 +1,198 @@
+import logging
+from time import time_ns
+from typing import Any, Dict, Optional
+
+from opentelemetry import trace as trace_api
+
+from mlflow.tracing.types.model import Event, Span, SpanContext, SpanType, Status, StatusCode
+
+_logger = logging.getLogger(__name__)
+
+
+class MLflowSpanWrapper:
+    """
+    A wrapper around OpenTelemetry's Span object to provide MLflow-specific functionality.
```

Comment:
Yes that works too. But generally composition is more flexible than inheritance in future modification ([ref](https://en.wikipedia.org/wiki/Composition_over_inheritance)). For example, if OpenTelemetry's span changes its method name, or we decide to switch to different backend, we can still keep the interface same. It might be super rare, but since this class is critical user-facing object, I'd like to take careful path here.

Of course it is Python so nothing prevent users from doing sth like `wrapper._span.xxx`, but at least it has underscore so gives an excuse for breaking change.

Reply:
I see, that's a good point! Though there're a lot functions we need expose 🥲

Reply:
I greatly prefer this approach as it will give us quite a bit of flexibility as we get feedback on the contents contained within traces.